service.xbmc.versioncheck/resources/lib/version_check/service.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+"""
+
+    Copyright (C) 2013-2014 Team-XBMC
+    Copyright (C) 2014-2019 Team Kodi
+
+    This file is part of service.xbmc.versioncheck
+
+    SPDX-License-Identifier: GPL-3.0-or-later
+    See LICENSES/GPL-3.0-or-later.txt for more information.
+
+"""
+
+import sys
+
+import xbmc  # pylint: disable=import-error
+import xbmcgui  # pylint: disable=import-error
+
+from .common import ADDON
+from .common import ADDON_NAME
+from .common import ADDON_VERSION
+from .common import dialog_yes_no
+from .common import localise
+from .common import log
+from .common import wait_for_abort
+from .common import message_restart
+from .common import message_upgrade_success
+from .common import upgrade_message
+from .common import upgrade_message2
+from .json_interface import get_version_file_list
+from .json_interface import get_installed_version
+from .versions import compare_version
+
+if sys.version_info[0] > 3 or (sys.version_info[0] == 3 and sys.version_info[1] >= 8):
+    import distro
```

Comment:
please see my bugreport: https://forum.kodi.tv/showthread.php?tid=160228&pid=2926800#pid2926800

Reply:
Looking into it

Reply:
I've added handling for when either platform or distro are missing.
zips are https://github.com/XBMC-Addons/service.xbmc.versioncheck/releases/tag/0.5.3 if you'd like to try it.

I don't want to add an external dependency to a service included with Kodi, should/could the distro module be included with v19?

Reply:
instead of adding it as a dependency, you could also include the `distro` module in the addon.

`platform` is part of the standard python library, so it should never be missing.

Reply:
I decided against it initially since a separate module is usually requested when that happens.
`distro` is only a single file plus license, I went ahead and included it. 

https://github.com/XBMC-Addons/service.xbmc.versioncheck/releases/tag/0.5.4